harness/api_checkers.go

```diff
@@ -0,0 +1,190 @@
+package harness
+
+import (
+	"bytes"
+	"testing"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/stretchr/testify/assert"
+)
+
+type CheckOpts struct {
+	CheckBaggageValues bool
+}
+
+func CheckStartSpan(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	span := tracer.StartSpan(
+		"Fry",
+		opentracing.Tag{Key: "birthday", Value: "August 14 1974"})
+	span.LogFields(
+		log.String("hospital", "Brooklyn Pre-Med Hospital"),
+		log.String("city", "Old New York"))
+	span.Finish()
+}
+
+func CheckStartSpanWithParent(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	parentSpan := tracer.StartSpan("parent")
+	assert.NotNil(t, parentSpan)
+
+	span := tracer.StartSpan(
+		"Leela",
+		opentracing.ChildOf(parentSpan.Context()))
+	span.Finish()
+
+	span = tracer.StartSpan(
+		"Leela",
+		opentracing.FollowsFrom(parentSpan.Context()),
+		opentracing.Tag{Key: "birthplace", Value: "sewers"})
+	span.Finish()
+
+	parentSpan.Finish()
+}
+
+func CheckSetOperationName(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	span := tracer.StartSpan("").SetOperationName("Farnsworth")
+	span.Finish()
+}
+
+func CheckSpanTagValueTypes(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	span := tracer.StartSpan("ManyTypes")
+	span.
+		SetTag("an_int", 9).
+		SetTag("a_bool", true).
+		SetTag("a_string", "aoeuidhtns")
+}
+
+func CheckSpanTagsWithChaining(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	span := tracer.StartSpan("Farnsworth")
+	span.
+		SetTag("birthday", "9 April, 2841").
+		SetTag("loves", "different lengths of wires")
+	span.
+		SetTag("unicode_val", "non-ascii: \u200b").
+		SetTag("unicode_key_\u200b", "ascii val")
+	span.Finish()
+}
+
+func CheckSpanLogs(t *testing.T, tracer opentracing.Tracer, opts CheckOpts) {
+	span := tracer.StartSpan("Fry")
+	span.LogKV(
+		"frozen.year", 1999,
+		"frozen.place", "Cryogenics Labs")
```

Comment:
given recent semantic specs I would've written this `event=frozen year=1999 place=xxx`